astropy/time/core.py

```diff
@@ -2228,12 +2272,29 @@ def to(self, unit, equivalencies=[]):
         return u.Quantity(self._time.jd1 + self._time.jd2,
                           u.day).to(unit, equivalencies=equivalencies)
 
-    def to_value(self, unit, equivalencies=[]):
-        """
-        The numerical value in the specified unit.
+    def to_value(self, *args, **kwargs):
+        """The numerical value in the specified format or unit.
+
+        With positional arguments, if the first argument is `None` or a format,
```

Comment:
OK, maybe this can fly if we are just very explicit about the API and keep it well-defined and simple.
```
This method is flexible and handles both conversion to a specified
``TimeDelta`` format / sub-format AND conversion to a specified unit.  If 
positional argument(s) are provided then the first one is checked to see if it is a valid
``TimeDelta`` format, and next it is checked to see if it is a valid unit or unit string.

To convert to a ``TimeDelta`` format and optional sub-format the options are::

  tm = TimeDelta(1.0 * u.s)
  tm.to_value('jd')  # equivalent of tm.jd
  tm.to_value('jd', 'decimal')  # convert to 'jd' as a Decimal object
  tm.to_value('jd', subfmt='decimal')
  tm.to_value(format='jd', subfmt='decimal')

To convert to a unit with optional equivalencies, the options are::

  tm.to_value('hr')  # convert to u.hr (hours)
  tm.to_value('hr', [])  # specify equivalencies as a positional arg
  tm.to_value('hr', equivalencies=[])
  tm.to_value(unit='hr', equivalencies=[])
```
I don't think that `None` should be allowed as the first arg, nor should we mix `subfmt` and `unit`.

Reply:
I'm happy to finish this off with an update to narrative docs and this docstring update.  Sound OK?

Reply:
Yes, definitely!  Sorry, have been working on other PRs...

Reply:
I pushed some doc updates.  Still need to put in examples and a couple more xrefs.

Reply:
More examples and xrefs done.  This is passing sphinx build locally, fingers crossed.